### PR TITLE
LF: Test evaluation order of lookup_by_key

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala [hunk 1]
```diff
@@ -889,7 +900,218 @@ class EvaluationOrderTest extends AnyFreeSpec with Matchers with Inside {
...
+          msgs shouldBe Seq("starts test", "maintainers")
+        }
+      }
+    }
+
+    "lookup_by_key" - {
+
+      "a non-cached global contract" - {
+
+        // TEST_EVIDENCE: Semantics: Evaluation order of successful lookup_by_key of a non-cached global contract
```
Comment (cocreature) [inline daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala:917]: is it worth testing both a lookup by key that returns `Some` and one that returns `None`?

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala [hunk 2]
```diff
@@ -889,7 +900,218 @@ class EvaluationOrderTest extends AnyFreeSpec with Matchers with Inside {
...
+            msgs shouldBe Seq(
+              "starts test",
+              "maintainers",
+              "queries key",
+              "ends test",
+            )
+          }
+        }
+
+        // TEST_EVIDENCE: Semantics: Evaluation order of lookup of a non-cached global contract with authorization failure
```
Comment (cocreature) [inline daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala:937]: should we also test visibilityt failures?

**Review by cocreature — APPROVED** [review]
Nice work thank you!

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala:981]: 5bcec92a99bd36858ff6f3435077d6c9acbe65d2

**Pull request merged**